typing/typecore.ml

```diff
@@ -5247,7 +5252,9 @@ and type_let ?check ?check_strict
                       {pat with pat_type =
                        snd (instance_poly ~keep_names:true false tl ty)}
                   | _ -> pat
-                in unify_pat (ref env) pat (type_approx env binding.pvb_expr))
+                in
+                let bound_expr = vb_exp_constraint binding in
```

Comment:
This is a bug fix in commit bc46f643f22127872c7: since the local type introduction is delayed, all accesses to `pvb_expr`  must go through calls to `vb_expr_constraint`. The code above was bypassing this step by using `binding.pvb_expr` when computing type approximation of recursive functions. Consequently, in 5.1.0~alpha1

```ocaml
let rec test : type a. a t -> a =
  function Int -> (1 : a)
 ```
 fails with
 
 > Unbound type constructor a
 
 because `binding.pvb_expr` lacks the `(type a) ->` prologue.

Reply:
I'm happy with this response, but can't seem to Resolve the conversation. Maybe I don't have enough permissions?